ggshield/core/git_shell.py

```diff
@@ -308,15 +312,15 @@ def tar_from_ref_and_filepaths(
 
     with tarfile.open(fileobj=tar_stream, mode="w:gz") as tar:
         for path in filepaths:
+            if accept_path is not None and not accept_path(path):
+                continue
 
-            def _readfile() -> str:
-                return read_git_file(ref, path, wd)
+            content = read_git_file(ref, path, wd)
```

Comment:
content should not be read if `accept_content` is `None`.

Reply:
We need to read the content regardless of the filters, to add it to the tar.